src/observer.ts

```diff
@@ -21,8 +21,14 @@ export class Observer {
   wsProvider: WsProvider
   api: ApiPromise
 
-  constructor(chain: OakChains) {
-    this.wsProvider = new WsProvider(OakChainWebsockets[chain])
+  /**
+   * constructor
+   * @param chain 
+   * @param options { providerUrl }, You can specify a custom provider url.
+   */
+  constructor(chain: OakChains, options: { providerUrl: string | undefined }) {
+    const { providerUrl } = options || {};
+    this.wsProvider = new WsProvider(providerUrl || OakChainWebsockets[chain])
```

Comment:
If you have to define `providerUrl || OakChainWebsockets[chain]` in both
`new Scheduler(chain, { providerUrl })` and `new Observer(chain, { providerUrl }),`, it’s better to define it outside for clarity.